From 4fa3a607fe8265d7fe0ca7f20c5eb5911e20a37f Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 8 Jul 2026 16:17:00 +0200
Subject: [PATCH 18/46] AnnaThermostat: add support for lan_state

Parse lan_state and publish it as an on/off switch sensor.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
Assisted-by: GPT-5.3-Codex

Upstream-Status: Inappropriate [WIP]
---
 hardware/AnnaThermostat.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/hardware/AnnaThermostat.cpp b/hardware/AnnaThermostat.cpp
index bc59d7a12..b425c739e 100644
--- a/hardware/AnnaThermostat.cpp
+++ b/hardware/AnnaThermostat.cpp
@@ -1143,6 +1143,14 @@ void CAnnaThermostat::GetMeterDetails()
 					SendPercentageSensor(applianceIDhex, 3, batterypercentage, maxlevel, ApplianceName + " " + sname);
 				}
 			}
+			else if (sname == "lan_state")
+			{
+				if (!tmpstr.empty())
+				{
+					bool isOn = (strcmp(tmpstr.c_str(), "up") == 0);
+					SendSwitch(applianceIDhex, 4, batterypercentage, isOn, 0, ApplianceName + " " + sname, m_Name);
+				}
+			}
             else if (sname == "central_heater_water_pressure")
             {
                 if (!tmpstr.empty())
